infra/Pos/Network/Policy.hs

```diff
@@ -0,0 +1,251 @@
+      ]
+    go (MsgMPC _) = [
+        EnqueueAll NodeCore (MaxAhead 1) PMedium
+        -- not sent to relay nodes
+      ]
+    go (MsgTransaction _) = [
+        EnqueueAll NodeCore (MaxAhead 20) PLow
+        -- not sent to relay nodes
+      ]
+
+-- | Default enqueue policy for relay nodes
+defaultEnqueuePolicyRelay :: EnqueuePolicy nid
+defaultEnqueuePolicyRelay = go
+  where
+    -- Enqueue policy for relay nodes
+    go :: EnqueuePolicy nid
+    go (MsgAnnounceBlockHeader _) = [
+        EnqueueAll NodeRelay (MaxAhead 0) PHighest
+      , EnqueueAll NodeCore  (MaxAhead 0) PHigh
+      , EnqueueAll NodeEdge  (MaxAhead 0) PMedium
+      ]
+    go MsgRequestBlockHeaders = [
+        EnqueueAll NodeCore  (MaxAhead 1) PHigh
+      , EnqueueAll NodeRelay (MaxAhead 1) PHigh
+      ]
+    go (MsgRequestBlocks _) = [
+        -- We never ask for blocks from edge nodes
+        EnqueueOne [NodeRelay, NodeCore] (MaxAhead 1) PHigh
+      ]
+    go (MsgTransaction _) = [
+        EnqueueAll NodeCore  (MaxAhead 20) PLow
+      , EnqueueAll NodeRelay (MaxAhead 20) PLow
+        -- transactions not forwarded to edge nodes
+      ]
+    go (MsgMPC _) = [
+        -- Relay nodes never sent any MPC messages to anyone
+      ]
+
+-- | Default enqueue policy for standard behind-NAT edge nodes
+defaultEnqueuePolicyEdgeBehindNat :: EnqueuePolicy nid
+defaultEnqueuePolicyEdgeBehindNat = go
+  where
+    -- Enqueue policy for edge nodes
+    go :: EnqueuePolicy nid
+    go (MsgTransaction OriginSender) = [
+        EnqueueAll NodeRelay (MaxAhead 1) PLow
+      ]
+    go (MsgTransaction (OriginForward _)) = [
+        -- don't forward transactions that weren't created at this node
+      ]
+    go (MsgAnnounceBlockHeader _) = [
+        -- not forwarded
+      ]
+    go MsgRequestBlockHeaders = [
+        EnqueueAll NodeRelay (MaxAhead 0) PHigh
+      ]
+    go (MsgRequestBlocks _) = [
+        -- Edge nodes can only talk to relay nodes
+        EnqueueOne [NodeRelay] (MaxAhead 0) PHigh
+      ]
+    go (MsgMPC _) = [
+        -- not relevant
+      ]
+
+-- | Default enqueue policy for exchange nodes
+defaultEnqueuePolicyEdgeExchange :: EnqueuePolicy nid
+defaultEnqueuePolicyEdgeExchange = go
+  where
+    -- Enqueue policy for edge nodes
+    go :: EnqueuePolicy nid
+    go (MsgTransaction OriginSender) = [
+        EnqueueAll NodeRelay (MaxAhead 6) PLow
+      ]
+    go (MsgTransaction (OriginForward _)) = [
+        -- don't forward transactions that weren't created at this node
+      ]
+    go (MsgAnnounceBlockHeader _) = [
+        -- not forwarded
+      ]
+    go MsgRequestBlockHeaders = [
+        EnqueueAll NodeRelay (MaxAhead 0) PHigh
+      ]
+    go (MsgRequestBlocks _) = [
+        -- Edge nodes can only talk to relay nodes
+        EnqueueOne [NodeRelay] (MaxAhead 0) PHigh
+      ]
+    go (MsgMPC _) = [
+        -- not relevant
+      ]
+
+-- | Default enqueue policy for edge nodes using P2P
+defaultEnqueuePolicyEdgeP2P :: EnqueuePolicy nid
```

Comment:
This is currently unused. In https://github.com/input-output-hk/cardano-sl/pull/1286 I defined a more fine grained default policy that took the entire topology into account. (https://github.com/input-output-hk/cardano-sl/pull/1286/files#diff-6c62a9f176e70074c6fb37cb9c6862bfR208)